### PR TITLE
replace StringBuffer with StringBuilder in local variable and delete …

### PR DESCRIPTION
**Pull request by AjaxXu**
…useless code

### Ⅰ. Describe what this PR did
1. replace StringBuffer with StringBuilder in local variable
2. remove useless init code in PreparedStatementProxy, which is duplicate in super class
3. Change some comment

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
unit test is cover by now

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/oracle/OracleUndoInsertExecutor.java [hunk 1]
```diff
@@ -47,9 +47,9 @@ protected String buildUndoSQL() {
             throw new ShouldNeverHappenException("Invalid UNDO LOG");
         }
         Row row = afterImageRows.get(0);
-        StringBuffer mainSQL = new StringBuffer("DELETE FROM " + keywordChecker.checkAndReplace(sqlUndoLog.getTableName()));
-        StringBuffer where = new StringBuffer(" WHERE ");
-        boolean first = true;
+        StringBuilder mainSQL = new StringBuilder("DELETE FROM " + keywordChecker.checkAndReplace(sqlUndoLog.getTableName()));
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/oracle/OracleUndoInsertExecutor.java]: append

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/oracle/OracleUndoUpdateExecutor.java [hunk 2]
```diff
@@ -44,8 +44,8 @@ protected String buildUndoSQL() {
             throw new ShouldNeverHappenException("Invalid UNDO LOG"); // TODO
         }
         Row row = beforeImageRows.get(0);
-        StringBuffer mainSQL = new StringBuffer("UPDATE " + keywordChecker.checkAndReplace(sqlUndoLog.getTableName()) + " SET ");
-        StringBuffer where = new StringBuffer(" WHERE ");
+        StringBuilder mainSQL = new StringBuilder("UPDATE " + keywordChecker.checkAndReplace(sqlUndoLog.getTableName()) + " SET ");
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/oracle/OracleUndoUpdateExecutor.java]: append

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/oracle/OracleUndoInsertExecutor.java (hunk 1 above)
Comment (AjaxXu) [reply rm-datasource/src/main/java/io/seata/rm/datasource/undo/oracle/OracleUndoInsertExecutor.java]: I can't get your point？Use append to replace +？

Comment (zjinlei) [reply rm-datasource/src/main/java/io/seata/rm/datasource/undo/oracle/OracleUndoInsertExecutor.java]: yes

**Pull request merged**